### PR TITLE
Add Sentence Transformer for text matching and Add readme

### PR DESCRIPTION
**Pull request by Steffy-zxf**
update text cls & text matching docs

### CODE REVIEW

File: PaddleNLP/examples/text_matching/simnet/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# 使用传统SimNet完成pointwise文本匹配任务
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/simnet/README.md]: 不要在SimNet前面+传统这个词

File: PaddleNLP/examples/text_matching/simnet/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+wget https://paddlenlp.bj.bcebos.com/data/simnet_word_dict.txt
+```
+
+我们以中文pointwise文本匹配数据集LCQMC为示例数据集，可以运行下面的命令，在训练集（train.tsv）上进行模型训练，并在开发集（dev.tsv）验证
+```shell
+# CPU启动
+python train.py --vocab_path='./simnet_word_dict.txt' --use_gpu=False --network=lstm --lr=5e-4 --batch_size=64 --epochs=5 --save_dir='./checkpoints'
+
+# GPU启动
+# CUDA_VSIBLE_DEVICES指定想要利用的GPU卡号，可以是单卡，也可以多卡
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/simnet/README.md]: 这句注释不用

File: PaddleNLP/examples/text_matching/simnet/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+# CUDA_VISIBLE_DEVICES=0 python train.py --vocab_path='./simnet_word_dict.txt' --use_gpu=True --network=lstm --lr=5e-4 --batch_size=64 --epochs=5 --save_dir='./checkpoints'
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/simnet/README.md]: 这里不应该“#“加注释

File: PaddleNLP/examples/text_matching/simnet/README.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+以上参数表示：
+
+* vocab_path: 词汇表文件路径。
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/simnet/README.md]: 所以args参数我们统一用`vocab_path`阔起来

File: PaddleNLP/examples/text_matching/simnet/train.py [hunk 5]
```diff
@@ -26,13 +26,13 @@
...
+parser.add_argument("--epochs", type=int, default=10, help="Number of epoches for training.")
+parser.add_argument('--use_gpu', type=eval, default=False, help="Whether use GPU for training, input should be True or False")
 parser.add_argument("--lr", type=float, default=5e-4, help="Learning rate used to train.")
 parser.add_argument("--save_dir", type=str, default='chekpoints/', help="Directory to save model checkpoint")
 parser.add_argument("--batch_size", type=int, default=64, help="Total examples' number of a batch for training.")
-parser.add_argument("--vocab_path", type=str, default="./data/term2id.dict", help="The directory to dataset.")
-parser.add_argument('--network', type=str, default="cnn", help="Which network you would like to choose bow, lstm, bilstm, gru, bigru, rnn, birnn, bilstm_attn and textcnn?")
+parser.add_argument("--vocab_path", type=str, default="./simnet_word_dict.txt", help="The directory to dataset.")
+parser.add_argument('--network', type=str, default="lstm", help="Which network you would like to choose bow, lstm, bilstm, gru, bigru, rnn, birnn, bilstm_attn and textcnn?")
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/simnet/train.py]: simnet没这么多选项了吧？

File: PaddleNLP/examples/text_matching/README.md [hunk 6]
```diff
@@ -1,5 +1,29 @@
...
+
+该项目展示了使用传统的[SimNet](./simnet) 和 [SentenceBert](./sentence_bert)两种方法完成pointwise本匹配任务。
+
+## Conventional Models
+
+[SimNet](./simnet) 展示了如何使用CNN、LSTM、GRU等网络完成pointwise文本匹配任务。
+
+## Pretrained Model (PTMs)
+
+[SentenceBert](./sentence_bert) 展示了如何使用以ERNIE为代表的模型Fine-tune完成pointwise文本匹配任务。
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/README.md]: 这类外部模型的，最后要价格todo，把相关论文的reference 和citation加上

File: PaddleNLP/examples/text_classification/rnn/README.md [hunk 7]
```diff
@@ -100,17 +90,17 @@ test_dataset = ppnlp.datasets.ChnSentiCorp('test')
...
 我们以中文情感分类公开数据集ChnSentiCorp为示例数据集，可以运行下面的命令，在训练集（train.tsv）上进行模型训练，并在开发集（dev.tsv）验证
 ```shell
 # CPU启动
-python train.py --vocab_path='./word_dict.txt' --use_gpu=False --network_name=bilstm --lr=5e-4 --batch_size=64 --epochs=5 --save_dir='./checkpoints'
+python train.py --vocab_path='./senta_word_dict.txt' --use_gpu=False --network_name=bilstm --lr=5e-4 --batch_size=64 --epochs=5 --save_dir='./checkpoints'
 
 # GPU启动
 # CUDA_VSIBLE_DEVICES指定想要利用的GPU卡号，可以是单卡，也可以多卡
-# CUDA_VISIBLE_DEVICES=0,1 python -m paddle.distributed.launch train.py --vocab_path='./word_dict.txt' --use_gpu=True --network_name=bilstm --lr=5e-4 --batch_size=64 --epochs=5 --save_dir='./checkpoints'
+# CUDA_VISIBLE_DEVICES=0 python train.py --vocab_path='./senta_word_dict.txt' --use_gpu=True --network_name=bilstm --lr=5e-4 --batch_size=64 --epochs=5 --save_dir='./checkpoints'
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_classification/rnn/README.md:107]: 为啥GPU启动要加#呢？

**Review by ZeyuChen — CHANGES_REQUESTED** [review]
细节文字意见修改

File: PaddleNLP/examples/text_classification/rnn/README.md (hunk 7 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_classification/rnn/README.md:107]: 因为前面有cpu启动方式

GPU启动方式只是给个示例作用

File: PaddleNLP/examples/text_matching/simnet/README.md (hunk 3 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/simnet/README.md]: 因为前面有cpu启动方式

GPU启动方式只是给个示例作用

Comment (ZeyuChen) [reply PaddleNLP/examples/text_matching/simnet/README.md]: 那是否可以采用分成两个代码block的方式说明呢？为什么GPU这里只是个示意？
CPU启动
```shell
xxxxxx
```
GPU启动
```shell
xxxxxx
```

File: PaddleNLP/examples/text_matching/simnet/README.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+**NOTE:** 如需恢复模型训练，则init_from_ckpt只需指定到文件名即可，不需要添加文件尾缀。如`--init_from_ckpt=checkpoints/0`即可，程序会自动加载模型参数`checkpoints/0.pdparams`，也会自动加载优化器状态`checkpoints/0.pdopt`。
+
+### 模型预测
+
+启动预测：
+```shell
+# CPU启动
+python predict.py --vocab_path='./simnet_word_dict.txt' --use_gpu=False --network=lstm --params_path=checkpoints/final.pdparams
+
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/simnet/README.md:147]: 建议分拆成两个code block

File: PaddleNLP/examples/text_matching/sentence_transformers/train.py [hunk 9]
```diff
@@ -0,0 +1,370 @@
...
+        learning_rate=lr_scheduler,
+        parameters=model.parameters(),
+        weight_decay=args.weight_decay,
+        apply_decay_param_fun=lambda x: x in [
+            p.name for n, p in model.named_parameters()
+            if not any(nd in n for nd in ["bias", "norm"])
+        ])
+
+    criterion = paddle.nn.loss.CrossEntropyLoss()
+    metric = paddle.metric.Accuracy(name='acc_accumulation')
```
Comment (ZeyuChen) [inline PaddleNLP/examples/text_matching/sentence_transformers/train.py]: 去掉这里的acc accumulate

File: PaddleNLP/examples/text_matching/simnet/README.md (hunk 1 above)
Comment (ZeyuChen) [reply PaddleNLP/examples/text_matching/simnet/README.md]: 去掉传统

**Review by ZeyuChen — CHANGES_REQUESTED** [review]
还有2个小的修改需要修改

File: PaddleNLP/examples/text_matching/simnet/README.md (hunk 1 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/simnet/README.md]: 已修改

File: PaddleNLP/examples/text_matching/simnet/README.md (hunk 2 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/simnet/README.md]: 已修改

File: PaddleNLP/examples/text_matching/simnet/README.md (hunk 3 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/simnet/README.md]: 已修改

File: PaddleNLP/examples/text_matching/simnet/README.md (hunk 4 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/simnet/README.md]: 已修改

File: PaddleNLP/examples/text_matching/simnet/train.py (hunk 5 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/simnet/train.py]: 已修改

File: PaddleNLP/examples/text_matching/README.md (hunk 6 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/README.md]: 已添加

File: PaddleNLP/examples/text_classification/rnn/README.md (hunk 7 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_classification/rnn/README.md:107]: 已修改

File: PaddleNLP/examples/text_matching/simnet/README.md (hunk 8 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/simnet/README.md:147]: 已修改

File: PaddleNLP/examples/text_matching/sentence_transformers/train.py (hunk 9 above)
Comment (Steffy-zxf) [reply PaddleNLP/examples/text_matching/sentence_transformers/train.py]: 已修改

**Review by ZeyuChen — APPROVED** [review]
LGTM
